src/main/java/com/odysseusinc/arachne/portal/util/UserUtils.java

```diff
@@ -33,14 +33,14 @@
 
 public class UserUtils {
     public static ArachneUser getCurrentUser() {
+        return getCurrentUser(SecurityContextHolder.getContext().getAuthentication());
+    }
 
-        ArachneUser user = null;
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        if (authentication.isAuthenticated()
-                && authentication.getPrincipal() instanceof ArachneUser) {
-            user = (ArachneUser) authentication.getPrincipal();
+    public static ArachneUser getCurrentUser(Authentication authentication) {
```

Comment:
can authentication be a null?

Reply:
Strictly speaking, depending on how springs-security is configured, for anonymous users you can have different behaviour: AnonymousAuthenticationToken or null. However it is not relevant, as all methods in question require authenticated user.